api-helpers/contentCreatorFunctions.ts

```diff
@@ -0,0 +1,156 @@
+import { resolve } from 'path';
+
+import Boom from '@hapi/boom';
+import cheerio from 'cheerio';
+import Slugify from 'slugify';
+
+import { openConnection } from './db';
+
+const NEVER = new Date(0);
+
+type BlogData = {
+  readonly name: string;
+  readonly href: string;
+  readonly favicon: string;
+  readonly feeds: readonly Feed[];
+};
+
+type Feed = {
+  readonly type: string;
+  readonly url: string;
+  readonly title: string;
+};
+
+export const addContentCreator = async (url: string, email?: string) => {
+  const prisma = await openConnection();
+  const { name, href, favicon, feeds } = await getBlogData(url);
+
+  const response = await prisma.blog.create({
+    data: {
+      name,
+      href,
+      favicon,
+      rss: feeds[0].url, // @to-do Ustalić, który jest najlepszy
+      slug: Slugify(name, { lower: true }),
+      lastUpdateDate: NEVER,
+    },
+  });
+};
+
+const getBlogData = async (url: string): Promise<BlogData> => {
+  const youtubeRss = getYoutubeRss(url);
+  if (youtubeRss !== false) {
+    return {
+      name: 'Some Youtube Channel', // @to-do Ustalić, nazwę strony z yt
+      href: url,
+      favicon: 'https://www.youtube.com/s/desktop/d743f786/img/favicon.ico',
+      feeds: [
+        {
+          type: '',
+          url: youtubeRss,
+          title: youtubeRss,
+        },
+      ],
+    };
+  } else {
+    const response = await fetch(url);
+    const htmlText = await response.text();
+    if (htmlText !== '') {
+      const $ = cheerio.load(htmlText);
+      const blogName = getBlogName($);
+      const favicon = getFavicon(url, $);
+      const feeds = searchFeed(url, $);
+      return {
+        name: blogName,
+        href: url,
+        favicon,
+        feeds,
+      };
+    }
+  }
+  throw Boom.badData();
+};
+
+const getYoutubeRss = (url: string) => {
+  const regex = RegExp(/^(http(s)?:\/\/)?((w){3}.)?youtu(be|.be)?(\.com)?\/.+/gm);
+  const isYoutubeUrl = regex.test(url);
+
+  if (isYoutubeUrl) {
+    const urlWithoutParams = url.split('?')[0];
+    let channel = '';
+    if (urlWithoutParams.split('channel/')[1]) {
+      channel = 'channel_id=' + urlWithoutParams.split('channel/')[1].split('/')[0];
+    } else if (url.split('user/')[1]) {
+      channel = 'user=' + urlWithoutParams.split('user/')[1].split('/')[0];
+    }
```

Comment:
Od 79 do 85 na pewno można zrobić lepiej używając modułu url + wywaliłbym to do funkcji